src/extensions/scratch3_mqtt/index.js

```diff
@@ -273,8 +272,8 @@ class Scratch3MQTTBlocks {
         this._client.on('connect',function(){
             console.log( "[mqtt] Connected !" );
         });
-        
-        this._client.on('error',function(){
+
+        this._client.on('error', () => {
```

Comment:
We'll need to use an arrow function here, those will automatically bind the `this` context inside the function to the one outside of it. 

If we use a regular function, the `this` context inside of our closure is a new context and doesn't have the `end()` function defined.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions